### DISCUSSION TITLE
Auth methods returning null on Android

### DISCUSSION THREAD

**Discussion by Nv7-GitHub** [pocketbase/pocketbase#3588] [Q&A]
I am using the pocketbase dart sdk. The `collections.create` method is returning `null` for our users collection, but `authWithPassword` also returns `null`. No record is created in the DB. We tried it on cellular data and WiFi, and both didn't work, no error was returned either. The app works on iOS, web, and android emulator, just not on Android physical devices.

It throws an error, which we catch, but the error is null so our handling code doesn't work.

**Comment by ganigeorgiev** [comment]
I'm not sure how to help you debug it.

The Dart SDK is built on top of the standard [dart-lang/http](https://github.com/dart-lang/http) and it should work just fine on all Dart platforms (_[on Web with some limitations](https://github.com/pocketbase/dart-sdk#limitations)_).

Are you sure that the SDK is actually making a request to the PocketBase server, aka. that it is reachable?

Does the request `Future` throws an error?

To inspect the request logs, you can check the logs in the _Admin UI > Logs_ and/or start the executable with `--debug`.

**Reply by Nv7-GitHub** [reply]
I don't think it is making a request because no record shows up in the dashboard. However, the phone should have access because it had Wifi and the app worked when opened on the browser, we tried this on multiple Android phones and all had the same issue

The request Future does throw an error, but the error is `null`

**Reply by Nv7-GitHub** [reply]
Here is the code:

```dart
on ClientException catch (e) {
        error = true;
        final data = e.response["data"];
        if (data.containsKey("email")) {
          emailError = data["email"]["message"];
        }
        if (data.containsKey("password")) {
          passwordError = data["password"]["message"];
        }
        if (data.containsKey("username")) {
          usernameError = data["username"]["message"];
        }
      }
```

**Reply by ganigeorgiev** [reply]
Could you show a more complete code-sample, eg. how do you initialize the SDK and how are you sending the request?

Also, what version of the PocketBase Dart SDK do you use?
Could you create a **minimal** reproducible repo?

To rule out PocketBase Dart SDK issue, you can try to send directly the request with the [`dart-lang/http` client](https://github.com/dart-lang/http/tree/master/pkgs/http) and see if the result would be different.

**Reply by ganigeorgiev** [reply]
Note that it could be also some issue with the permissions of your Android app (aka. check your manifest file) as they usually differ from those on iOS.

See also https://docs.flutter.dev/data-and-backend/networking.

**Reply by Nv7-GitHub** [reply]
We have the networking permission correct, but I will try using the dart-lang http client to see what happens.

**Reply by ganigeorgiev** [reply]
You can also try to remove the `on ClientException` part from the `try/catch` block to catch all errors as the exception could be thrown from somewhere else if there is other code in the try block.
